### ISSUE TITLE
[Bug] Missing shape infer for conv2d

### ISSUE THREAD

**Issue by zhouyuan1119** [awslabs/ratex#14]
Note: This bug is triggered by an unconventional use of Ratex. Normally what I am describing below should not happen. 

Description
---
I was trying to inspect the lazy tensor IR of the LeNet example **without** using `ratex.jit.script`. I was expecting that the lazy tensor IR would contain all PyTorch operators in fwd, bwd, and weight update, but the example crashed before generating the lazy tensor IR graph. More precisely, it crashed during shape inference when trying to lower to RAF:
```
RuntimeError: [22:29:54] /home/ubuntu/ratex/ratex/ratex/csrc/compiler/raf_node_lowering.cpp:1374: Shape inference not supported for operator: aten::convolution_overrideable
```
The `convolution_overrideable` operator is not listed in the function and thus falls to the default case. Seems like we are only supporting a small set of native PyTorch operators over here, which is sufficient if we only consider ops used during weight update but will error out for many ops inside the fwd/bwd passes. 

Tentative Solution
---
The solution is conceptually simple, that is, to add the missing cases for all other operators. My concern here is that whether this is necessary, because this won't be triggered if we use Ratex normally. Please let me know if you have any suggestions.

**Comment by zachzzc** [comment]
We will have a better support for the ops after Awais's work @anadeem2. He will soon add the support for matmul and other operators that are used in Bert-like model. The support for CV model might still need help.

**Comment by zhouyuan1119** [comment]
Addressed in PR #16.
